ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
+    const createButton = screen.getByRole('button', { name: 'Create' });
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(150);
+    // This should make the cost about $50 per hour.
+    await pickNumWorkers(200);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should prevent runtime creation when worker count is invalid', async () => {
+    setCurrentRuntime(null);
+    const { container } = await component();
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    const createButton = screen.getByRole('button', { name: 'Create' });
+    await pickNumWorkers(0);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(1);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should allow runtime creation when running cost is too high for user provided billing', async () => {
+    currentWorkspaceStore.next({
+      ...workspaceStubs[0],
+      accessLevel: WorkspaceAccessLevel.OWNER,
+      billingAccountName: 'user provided billing',
+    });
+
+    setCurrentRuntime(null);
+    const { container } = await component();
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    // This should make the cost about $50 per hour.
+    await pickNumWorkers(20000);
+    const createButton = screen.getByRole('button', { name: 'Create' });
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should prevent runtime creation when running cost is too high for paid tier', async () => {
+    setCurrentRuntime(null);
+    currentWorkspaceStore.next({
+      ...workspaceStubs[0],
+      accessLevel: WorkspaceAccessLevel.WRITER,
+      cdrVersionId: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID,
+    });
+    const { container } = await component();
+
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    // This should make the cost about $140 per hour.
+    await pickNumWorkers(600);
+    let createButton = screen.getByRole('button', { name: 'Create' });
+    expectButtonElementEnabled(createButton);
+
+    // This should make the cost around $160 per hour.
+    await pickNumWorkers(700);
+    createButton = screen.getByRole('button', { name: 'Create' });
+    // We don't want to disable for user provided billing. Just put a warning.
+    expectButtonElementEnabled(createButton);
+    expect(
+      screen.findByText(
+        'Your runtime is expensive. Are you sure you wish to proceed?'
+      )
+    ).toBeTruthy();
+
+    await pickNumWorkers(2);
```

Comment:
not sure why this part of the test is needed

Reply:
I updated it to say if the number of worker is within range remove the banner. If that condition doesnt make sense i can remove it

Reply:
Basically saying the banner does not stick and can be reset

Reply:
ok great, this works